lib/wallet/rpc.js

```diff
@@ -1170,38 +1172,187 @@ class RPC extends RPCBase {
   }
 
   async listTransactions(args, help) {
+    throw new Error('Deprecated: `listtransactions`. ' +
+                    'Use `listhistory` and related methods.');
+  }
+
+  async listHistory(args, help) {
     if (help || args.length > 4) {
       throw new RPCError(errs.MISC_ERROR,
-        'listtransactions ( "account" count from includeWatchonly)');
+        'listhistory "account" ( limit, reverse )');
     }
 
     const wallet = this.wallet;
     const valid = new Validator(args);
-    let name = valid.str(0);
-    const count = valid.u32(1, 10);
-    const from = valid.u32(2, 0);
-    const watchOnly = valid.bool(3, false);
+    const name = valid.str(0);
+    const limit = valid.u32(1, 10);
+    const reverse = valid.bool(2, false);
 
-    if (wallet.watchOnly !== watchOnly)
-      return [];
+    if (limit > this.maxTxs)
+      throw new RPCError(errs.INVALID_PARAMETER,
+                         `Limit above max of ${this.maxTxs}.`);
 
-    if (name === '')
-      name = 'default';
+    const recs = await wallet.listHistory(name, {limit, reverse});
 
-    const txs = await wallet.getHistory();
+    const out = [];
+    for (let i = 0; i < recs.length; i++)
+      out.push(await this._toListTX(recs[i]));
+
+    return out;
+  }
 
-    common.sortTX(txs);
+  async listHistoryAfter(args, help) {
+    if (help || args.length > 4) {
+      throw new RPCError(errs.MISC_ERROR,
+        'listhistoryafter "account", "txid" ( limit, reverse )');
+    }
+
+    const wallet = this.wallet;
+    const valid = new Validator(args);
+    const name = valid.str(0);
+    const txid = valid.buf(1);
+    const limit = valid.u32(2, 10);
+    const reverse = valid.bool(3, false);
+
+    if (limit > this.maxTxs)
+      throw new RPCError(errs.INVALID_PARAMETER,
+                         `Limit above max of ${this.maxTxs}.`);
+
+    const recs = await wallet.listHistoryAfter(name, {
+      txid: txid.reverse(),
+      limit,
+      reverse
+    });
 
-    const end = from + count;
-    const to = Math.min(end, txs.length);
     const out = [];
+    for (let i = 0; i < recs.length; i++)
+      out.push(await this._toListTX(recs[i]));
 
-    for (let i = from; i < to; i++) {
-      const wtx = txs[i];
-      const json = await this._toListTX(wtx);
-      out.push(json);
+    return out;
+  }
+
+  async listHistoryByTime(args, help) {
+    if (help || args.length > 4) {
+      throw new RPCError(errs.MISC_ERROR,
+        'listhistorybytime "account", "timestamp" ( limit, reverse )');
+    }
+
+    const wallet = this.wallet;
+    const valid = new Validator(args);
+    const name = valid.str(0);
+    const timestamp = valid.str(1);
+    const limit = valid.u32(2, 10);
+    const reverse = valid.bool(3, false);
+
+    if (limit > this.maxTxs)
+      throw new RPCError(errs.INVALID_PARAMETER,
+                         `Limit above max of ${this.maxTxs}.`);
+
+    const date = new Date(timestamp);
+    /* eslint eqeqeq: "off" */
```

Comment:
Any particular reason to use `==` here over `===`?

Reply:
Yeah so it's compared as a string, see:
```
> const d = new Date(NaN);
> d == 'Invalid Date'
true
> d === 'Invalid Date'
false
```

Reply:
Could also do: `date.toString() === 'Invalid Date'`